### PR TITLE
feat: ranking panel rework

### PR DESCRIPTION
**Pull request by VIKTORVAV99**
## Issue
Closes: AVO-17
Closes: AVO-337
Closes: #4530

## Description

Link to Figma: https://www.figma.com/design/O1DtS6UyZHSH4IymG8Ufsw/%F0%9F%8E%B9-Design-Hack?node-id=4195-2114&t=G7O58Pa8jUXOwk6y-1
### Double check

- [ ] I have tested my parser changes locally with `poetry run test_parser "zone_key"`
- [ ] I have run `pnpx prettier@2 --write .` and `poetry run format` in the top level directory to format my changes.

### CODE REVIEW

File: web/src/components/Button.tsx
Comment (madsnedergaard) [inline web/src/components/Button.tsx:189]: Not a big fan of putting all of these in the Button.tsx file :)
To me they are not a "basic" component, and thus should live elsewhere. 
Perhaps these (and potentially more) could live under a new feature-folder called something like `/features/about`? The idea being to group components that are part of the same "feature", despite being used in different places (panel and modal).

Comment (VIKTORVAV99) [reply web/src/components/Button.tsx:189]: I agree in principle but nothing really felt right so I put them in button, at least for now.

I'll happily move them once everything is ready to be reviewed 👍🏼

Currently I'm holding off on doing the last style changes until #6855 is merged.

Comment (VIKTORVAV99) [reply web/src/components/Button.tsx:189]: @madsnedergaard how would you feel about putting it in a buttons sub folder in components and splitting it out in individual files?

I think that might be the most extensible option?

Comment (madsnedergaard) [reply web/src/components/Button.tsx]: I personally don't feel like they are "core components" used to build up features, but we're also getting into nitty-gritty details so it's not something we need to argue too much about 😄

File: web/src/components/Button.tsx
Comment (VIKTORVAV99) [inline web/src/components/Button.tsx:9]: There was a ESLint rule that complained about this when I tried to upgrade it, figured I might as well change it now rather than later but I'd be happy to change it back if you think it should be a separate PR.

**Comment by Alportan** [comment]
Looking good Viktor! 🙌 

I added a few [comments in Figma](https://www.figma.com/design/O1DtS6UyZHSH4IymG8Ufsw/%F0%9F%8E%B9-Design-Hack?node-id=4884-15482&t=2sUaQO9bTUjkBge0-1) here as we agreed, + a few extra thoughts for future improvements that Tony and Mads mentioned.

**Comment by tonypls** [comment]
The padding for each of these lines would be nice if it was equal. Slightly more than the top padding is currently:
Screenshot 2024-07-31 at 10 53 44 AM


Optional fix: iPad mini or smaller desktop size the buttons are not completely visible
image

Will also do a code review now (and ask the pr agent to do one 🤖)

/review

**Comment by VIKTORVAV99** [comment]
> The padding for each of these lines would be nice if it was equal. Slightly more than the top padding is currently: Screenshot 2024-07-31 at 10 53 44 AM
> 
> Optional fix: iPad mini or smaller desktop size the buttons are not completely visible image
> 
> Will also do a code review now (and ask the pr agent to do one 🤖)
> 
> /review

Overlap should be fixed or mitigated by #7024 which was originally part of #6911.

I'll look into the padding and double check it with Alex as well. 👍🏻

**Comment by VIKTORVAV99** [comment]
Fixed the margins and hight so it looks more uniform but a more permanent responsive fix requires https://github.com/electricitymaps/electricitymaps-contrib/pull/7020 or something similar.

Possibly also some responsive tweaks about the header as it looks very strange on say a iPad at the moment (when visiting the website).

**Review by tonypls — COMMENTED** [review]
Couple of small comments but overall loooks great! 

Feel free to scrutinize my padding changes, my goal was to have no padding on the Outer left panel section, reduce the padding on the new stuff to make it maximize ranking space and just meet the top of the Historical Slider component.

File: web/src/features/modals/InfoModal.tsx
Comment (VIKTORVAV99) [reply web/src/features/modals/InfoModal.tsx:29]: Absolutly nothing it seems 😅

I'll remove it.

File: web/src/features/modals/InfoModal.tsx
Comment (VIKTORVAV99) [reply web/src/features/modals/InfoModal.tsx:31]: I'll adjust it slightly but it's not meant to be super visible. Let me know what you think about the new version.

**Comment by tonypls** [comment]
Do you know where this border on the input came from? 
image

**Comment by VIKTORVAV99** [comment]
> Do you know where this border on the input came from? It appeared after my commit but I can't figure out how?! image

It has appeared in other PRs as well, best I can think of is some sort of dependency change that happened earlier today (Dependabot updates all dependencies the 1st of each month) but I'll look into why it's happening.

It has nothing to do with this PR though.

**Comment by tonypls** [comment]
> > Do you know where this border on the input came from? It appeared after my commit but I can't figure out how?! image
> 
> It has appeared in other PRs as well, best I can think of is some sort of dependency change that happened earlier today (Dependabot updates all dependencies the 1st of each month) but I'll look into why it's happening.
> 
> It has nothing to do with this PR though.
> 
> EDIT: It's not affected on staging so probably not a dependency change but it's also affecting #7048 so I don't really know what the issue is. Will try and dig into it a bit.

I'm very confused because it's doesn't happen with this branch locally but it does on the preview 🤔

**Review by tonypls — APPROVED** [review]
LGTM 🎉 

Looks great, glad to get this one out there :)

**Comment by VIKTORVAV99** [comment]
> > > Do you know where this border on the input came from? It appeared after my commit but I can't figure out how?! image
> > 
> > 
> > It has appeared in other PRs as well, best I can think of is some sort of dependency change that happened earlier today (Dependabot updates all dependencies the 1st of each month) but I'll look into why it's happening.
> > It has nothing to do with this PR though.
> > EDIT: It's not affected on staging so probably not a dependency change but it's also affecting #7048 so I don't really know what the issue is. Will try and dig into it a bit.
> 
> I'm very confused because it's doesn't happen with this branch locally but it does on the preview 🤔

If you try and reinstall your dependencies does it show up then? Maybe it's a dependency thing anyway (I forgot that I now have to update staging as well so maybe it's there anyway).

**Comment by VIKTORVAV99** [comment]
Narrowed down the problem to this PR https://github.com/tailwindlabs/tailwindcss-forms/pull/141 of `@tailwindcss/forms` which was included in version `0.5.4`.

I'll create a PR to revert the package here and then open an issue with them describing the issue.

And if this is the intended behaviour (we were not specifying a type) then I'll create a pr that removes this styling instead and add `type='search` to the input for accessibility purposes.

**Comment by tonypls** [comment]
> Narrowed down the problem to this PR [tailwindlabs/tailwindcss-forms#141](https://github.com/tailwindlabs/tailwindcss-forms/pull/141) of `@tailwindcss/forms` which was included in version `0.5.4`.
> 
> I'll create a PR to revert the package here and then open an issue with them describing the issue.
> 
> And if this is the intended behaviour (we were not specifying a type) then I'll create a pr that removes this styling instead and add `type='search` to the input for accessibility purposes.

Nice investigating! Thanks

**Pull request merged**
